lockc/src/bpfstructs.rs

```diff
@@ -5,64 +5,43 @@
 #![allow(non_snake_case)]
 include!(concat!(env!("OUT_DIR"), "/bindings.rs"));
 
-use byteorder::{NativeEndian, WriteBytesExt};
+use std::ffi::CString;
 
 #[derive(thiserror::Error, Debug)]
 pub enum NewBpfstructError {
     #[error("FFI nul error")]
     NulError(#[from] std::ffi::NulError),
 }
 
-#[derive(thiserror::Error, Debug)]
-pub enum MapOperationError {
-    #[error("could not convert the key to a byte array")]
-    ByteWriteError(#[from] std::io::Error),
-
-    #[error("libbpf error")]
-    LibbpfError(#[from] libbpf_rs::Error),
-}
-
-/// Deletes an entry from the given map under the given key.
-pub fn map_delete(map: &mut libbpf_rs::Map, key: u32) -> Result<(), MapOperationError> {
-    let mut key_b = vec![];
-    key_b.write_u32::<NativeEndian>(key)?;
-
-    map.delete(&key_b)?;
-
-    Ok(())
-}
-
-pub trait BpfStruct {
-    /// Updates the given map with an entry under the given key and a value
-    /// with a binary representation of the struct.
-    fn map_update(&self, map: &mut libbpf_rs::Map, key: u32) -> Result<(), MapOperationError> {
-        let mut key_b = vec![];
-        key_b.write_u32::<NativeEndian>(key)?;
-
-        let val_b = unsafe { plain::as_bytes(self) };
-
-        map.update(&key_b, val_b, libbpf_rs::MapFlags::empty())?;
-
-        Ok(())
-    }
-}
-
-impl BpfStruct for container {}
-impl BpfStruct for process {}
-impl BpfStruct for accessed_path {}
-
 impl accessed_path {
     /// Creates a new accessed_path instance and converts the given Rust string
     /// into C fixed-size char array.
     pub fn new(path: &str) -> Result<Self, NewBpfstructError> {
-        let mut path_b = std::ffi::CString::new(path)?.into_bytes_with_nul();
+        let mut path_b = CString::new(path)?.into_bytes_with_nul();
         path_b.resize(PATH_LEN as usize, 0);
         Ok(accessed_path {
             path: path_b.try_into().unwrap(),
         })
     }
 }
 
+impl container_id {
+    /// Creates a new container_id instance and converts the given Rust string
+    /// into C fixed size char array.
+    pub fn new(id: &str) -> Result<Self, NewBpfstructError> {
+        let mut id_b = CString::new(id)?.into_bytes_with_nul();
+        id_b.resize(CONTAINER_ID_LIMIT as usize, 0);
+        Ok(container_id {
+            id: id_b.try_into().unwrap(),
```

Comment:
You should not `unwrap` here, rather bubble up the error. This function already returns a `Result`

Reply:
Yes, this is what I would do. Eventually define a new error type with `thiserror` and then rely on`map_err` to do the conversion